tasks/main-clevis.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Ensure required packages are installed
+  package:
+    name: "{{ __nbde_client_package_list }}"
+    state: present
+
+
+- name: Ensure we transfer keyfiles
+  copy:
+    src: "{{ item.1.keyfile | default('/dev/null') }}"
+    dest: "{{ nbde_client_remote_dir }}/"
+    mode: '0400'
+  with_subelements:
+    - "{{ nbde_client_bindings }}"
+    - devices
+
+
+- name: Perform clevis operations
+  nbde_client_clevis:
+    bindings: "{{ nbde_client_bindings | default([]) }}"
+    data_dir: "{{ nbde_client_remote_dir }}"
+  notify: update initramfs
+
+
+- name: Ensure we dispose of any transferred keyfiles
```

Comment:
Is this a task you want to perform even if `name: Perform clevis operations` fails?  If so, suggest you use a `block/rescue/always` e.g. put the cleanup in an `always` block.

Reply:
Good point. I will do this.